include/drjit-core/jit.h

```diff
@@ -2474,6 +2480,153 @@ extern JIT_EXPORT uint32_t jit_array_write(uint32_t target, uint32_t offset,
 extern JIT_EXPORT uint32_t jit_array_read(uint32_t source, uint32_t offset,
                                           uint32_t mask);
 
+// ====================================================================
+//     Core functionality to enable kernel freezing via @dr.freeze
+// ====================================================================
+
+/// Opaque data structure, storing the recording of a thread state
```

Comment:
"[..] storing a sequence of Dr.Jit operations recorded using the API below" (The thread state itself isn't really recorded, it's more how the thread state is used)